File: Programmers(프로그래머스)/LV1/Java/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        
          int sum = 0;
	      int total = x;
	   
	      while(x>0)
	      {
	         sum+= x%10;
	         x=x/10;
	      }
	      
	    boolean answer = (total % sum == 0)?true:false;
        
        return answer;
    }
}

File: Programmers(프로그래머스)/LV1/Python/하샤드 수.py
def solution(x):
      answer = True
      a = []

      for i in str(x):
            a.append(i)

      for i in range(0, len(a)) :
            a[i] = int(a[i])
      
      sumA = sum(a)

      if x % sumA != 0 :
            answer = False

      return answer